### PR TITLE
Setup/add pre commit

### PR DESCRIPTION
**Pull request by johnnv1**
Add pre-commit config  `.pre-commit-config.yaml` to use [pre-commit](https://github.com/pre-commit/pre-commit)

Hooks added:
-   check-added-large-files
-   check-merge-conflict
-   check-case-conflict
-   check-yaml
-   detect-private-key
-   debug-statements
-   name-tests-test
-   end-of-file-fixer
-   trailing-whitespace
-   double-quote-string-fixer
-   requirements-txt-fixer
-   flake8
-   autopep8
-   pyupgrade
-   reorder-python-imports
-   setup-cfg-fmt
-   mypy

Also update the file to matches with all refactoring hooks and pep8 patterns.

At the end I have decided to **drop python 3.6 support**

### CODE REVIEW

**Comment by johnnv1** [comment]
For some reason, my commit msg/title is missing the name of the hooks

Commits: 
- [8e6383a5b5b3e5eddf7de18d9e5618fbef381e11](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/8e6383a5b5b3e5eddf7de18d9e5618fbef381e11) - add `trailing-whitespace` hook
- [bb3523f6dc98662bf60287e02dd624bdb22a1ac6](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/bb3523f6dc98662bf60287e02dd624bdb22a1ac6) - add `double-quote-string-fixer` hook
- [b3d94b065419c0ab8c18bdd0053fabe9ae32501e](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/b3d94b065419c0ab8c18bdd0053fabe9ae32501e) - add `requirements-txt-fixer` hook
- [ddad31fdff39ce1c65a470c939a5bef9b895bc01](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/ddad31fdff39ce1c65a470c939a5bef9b895bc01) - add `autopep8` hook
- [df98ca913392458a79f44025e5b0a0e0dbbc66e7](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/df98ca913392458a79f44025e5b0a0e0dbbc66e7) - add `pyupgrade` hook
- [1e6e7ac2b6a919596a239feeb923eeed89844295](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/1e6e7ac2b6a919596a239feeb923eeed89844295) - add `setup-cfg-fmt` hook
- [624d18bda455c3d949ab18e1c8d7e7faae89f1ac](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/624d18bda455c3d949ab18e1c8d7e7faae89f1ac) - add `reorder-python-imports` hook and drop python 3.6 supports
- [163f544de46a6996d32bab5b6da4a15c19bc77d0](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/163f544de46a6996d32bab5b6da4a15c19bc77d0) - fixes by `pyupgrade` hook (without py36 support)
- [23f0d6a46dd29e1fc5e37744151c0d8d9c5fc275](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/23f0d6a46dd29e1fc5e37744151c0d8d9c5fc275) - add `mypy` hook
- [403ba056fb8edde54b9c4d2228dc021f2f207c49](https://github.com/lapix-ufsc/lapixdl/pull/13/commits/403ba056fb8edde54b9c4d2228dc021f2f207c49) - add `flake8` hook

File: examples.py [hunk 1]
```diff
@@ -85,7 +91,7 @@ def evaluate_detection_example():
     classes = ['kite', 'person']
 
     # Image shape
-    mask_shape = (480, 640)
+    # mask_shape = (480, 640)
```
Comment (Andrevmatias) [inline examples.py:94]: Why did you comment some lines of code? If they are unused, I think we should delete them.

File: setup.cfg [hunk 2]
```diff
@@ -7,21 +7,29 @@ long_description_content_type = text/markdown
...
+    matplotlib
     numpy
     pandas
-    tqdm
     seaborn
-    pandas
-    matplotlib
-
+    tqdm
 python_requires = >=3.6
```
Comment (Andrevmatias) [inline setup.cfg:30]:
```suggestion
python_requires = >=3.7
```

File: setup.cfg [hunk 3]
```diff
@@ -7,21 +7,29 @@ long_description_content_type = text/markdown
...
 
 [options.packages.find]
 exclude =
     tests*
-    data*
```
Comment (Andrevmatias) [inline setup.cfg:27]: Why did you include `data`?

**Review by Andrevmatias — CHANGES_REQUESTED** [review]
Also, codefactor pointed an unnecessary else statement in model.py

File: setup.cfg (hunk 3 above)
Comment (johnnv1) [reply setup.cfg:27]: I just forgot to delete, can delete this

**Comment by johnnv1** [comment]
> Also, codefactor pointed an unnecessary else statement in model.py

File: setup.cfg (hunk 3 above)
Comment (Andrevmatias) [reply setup.cfg:27]: I got it wrong. You already deleted it, it is OK.

File: lapixdl/evaluation/model.py [hunk 4]
```diff
@@ -614,22 +618,22 @@ def iou(self) -> float:
...
             Uses all points if None. Defaults to None.
 
         Returns:
             float: Average Precision metric.
         """
-        if not interpolation_points is None:
+        if interpolation_points is not None:
             return self._interpolated_average_precision(interpolation_points)
         else:
             return self._average_precision()
```
Comment (Andrevmatias) [inline lapixdl/evaluation/model.py:639]: I think this is what codefactor want you to do:
```suggestion
            return self._interpolated_average_precision(interpolation_points)
        return self._average_precision()
```

If you think it is better to keep the "else" statement, I think we should disable the `no-else-return` rule.

**Review by johnnv1 — COMMENTED** [review]
.

**Review by Andrevmatias — APPROVED** [review]

**Pull request merged**